app/utils/validation.ts

```diff
@@ -109,23 +113,59 @@ export const ifNotField =
     context[field] ? ([true] as const) : validator(value, context);
 
 export function createValidator(
-  fieldValidators: { [field: string]: ReturnType<Validator>[] },
-  rawValidator?: (input) => ValidatorResult
+  fieldValidators: {
+    [field: string]: (ReturnType<Validator> | ReturnType<AsyncValidator>)[];
+  },
+  rawValidator?: (input) => ValidatorResult,
+  makeAsync = false
```

Comment:
```suggestion
  async = false
```
a so called smør-på-flesk-variabel